### DISCUSSION TITLE
How to redirect folders in Home

### DISCUSSION THREAD

**Discussion by doobes** [elementary/files#2114] [Q&A]
Hi there,

Fairly newby to EOS.  I've done some spelunking around and the answers to this question are so old they no longer work.

I'm on Jolnir, so here goes:

![files](https://user-images.githubusercontent.com/23409761/210879748-f83269ec-f8c5-4157-bdb1-d44e9c6afed2.jpeg)

I've got a separate hard drive with files that I share with a Windows boot.  How do I point the various folders in Home to that drive?

thanks,

Chris

**Comment by jeremypw** [comment]
One way is to edit `~/.config/user-dirs.dirs` (where `~` is your home directory).  This file includes instructions for editing the paths to your standard home directories.

Or you can just replace the existing (empty) folders in your home with links to the desired folders.  Just use "Copy as Link" and "Paste Link" context menu options within Files and rename to the original home folder name.

The partition containing the desired folders should be always mounted on startup by being listed in `/etc/fstab`.

**Comment by doobes** [comment]
Thanks Jeremy

I've tried modifying ~/.config/user-dirs.dirs.

As a trial, I changed

XDG_DOCUMENTS_DIR="$HOME/Documents"

to

XDG_DOCUMENTS_DIR="/media/chris/01D62334F6AEF120/Documents"

When EOS restarts, it overwrites this with 

XDG_DOCUMENTS_DIR="$HOME/"

And also the Documents folder on the drive is now inaccessible.

So I reset ~/.config/user-dirs.dirs to original.

What I did do was to create bookmarks to the appropriate folders on the data drive.  These don't show up in "Home", but for now this will suffice.

Thanks

**Comment by doobes** [comment]
Well, editing ~/.config/user-dirs.dirs most assuredly does not change the mapping for the folders.

Realized my issue with the above was the mount label changed at every boot.

So I mounted  the internal drive via an fstab entry (/mnt/d-drive) and then changed ~/.config/user-dirs.dirs to look like this:

XDG_DESKTOP_DIR="$HOME/Desktop"
XDG_DOWNLOAD_DIR="/mnt/d-drive/Downloads"
XDG_PUBLICSHARE_DIR="$HOME/Public"
XDG_DOCUMENTS_DIR="/mnt/d-drive/Documents"
XDG_MUSIC_DIR="/mnt/music"
XDG_PICTURES_DIR="/mnt/d-drive/Pictures"
XDG_VIDEOS_DIR="/mnt/video"
XDG_TEMPLATES_DIR="$HOME/Templates"

Clicking on the appropriate folder link in files results in either the original location when the mapping was to an internal drive, or a no connection when the mapping was for my NSA box (/mnt/music,/mnt/video, etc).

**Reply by jeremypw** [reply]
Interesting. I assume if you navigate in Files to e.g. `/mnt/d-drive/Pictures` you *can* see your files?  I admit my partitions are mounted to `/media/ but I would not have thought that would make a difference.

It occurs to me that you  may need to delete (or rename) `~/.config/gtk-3.0/bookmarks` so that the sidebar bookmarks are refreshed. Sorry I forgot that.

EDIT:  I tried it on my system and can confirm you need to delete (or manually amend) the paths in the `~/.config/gtk-3.0/bookmarks` file for Files to reload bookmarks from `user-dirs.dirs`

I am not sure which file other apps like Chromium look at to determine e.g. the Download directory.

**Reply by doobes** [reply]
NT

**Comment by jeremypw** [comment]
I guess it might be nice for Files to compare the two files when it starts to pick up changes in `user-dirs.dirs` and synchronise them with the bookmarks file.  Not sure if other filemanagers do this.  You could raise an issue about it.

**Comment by doobes** [comment]
Well,

I'm genuinely not certain we gained much.

Yes I can see the appropriate drives/folder contents in the /mnt/ folder.  Just for fun, I relocated all of them to the /media/ folder/. The most immediate result is the way the mounted folders are represented in the files app.  I can no longer right click on the mounts and create bookmarks.

I edited ~/.config/user-dirs.dirs as such:

XDG_DESKTOP_DIR="$HOME/Desktop"
XDG_DOWNLOAD_DIR="/media/d-drive/Downloads"
XDG_PUBLICSHARE_DIR="$HOME/Public"
XDG_DOCUMENTS_DIR="/media/d-drive/Documents"
XDG_MUSIC_DIR="/media/music"
XDG_PICTURES_DIR="/media/d-drive/Pictures"
XDG_VIDEOS_DIR="/media/video"
XDG_TEMPLATES_DIR="$HOME/Templates"
XDG_LIBRARY_DIR="/media/library"
XDG_PROJECTS_DIR="/media/projects"
XDG_PHOTOS_DIR="/media/photos"

I deleted the bookmarks file.

Rebooting this is what I get:

![files](https://user-images.githubusercontent.com/23409761/211212176-19979107-2040-4946-a677-70d8a660f642.jpeg)

The folders in the Home folder (Desktop, Documents, Downloads, etc, etc) still all point to /home/chris/

The icons that are shown below the Home icon connect to the items to the places I want.

Several selections I created in ~/.config/user-dirs.dirs don't display.

Still not getting what I want to do. I want the folders in the Home folder to point to where I want them, not /homes/chris....

Why is this so difficult?

Thanks,

**Comment by jeremypw** [comment]
If you do not want the partition(s) containing your home folders to appear as Storage then you should put them in `/mnt` as before. In my case, I have all my home data in one partition so its not a problem - I only get one extra entry in Storage. In your case, each home subfolder seems to have its own partition presumably because of the way you constructed the fstab file - is that necessary?  Not sure why your Windows partitions have a "remote" icon - if they are on the same computer they should have a "disk" icon - what path do they have?  You should be able to bookmark anything that can be displayed in the view by right clicking on the view.

Check the actual path of the desired home subfolders in `/media` (look at the tooltip when you hover the partition entry).  In my case the are subfolders of `/media/<username>/` not `/media`. 

I agree it is fiddly to setup - all the paths in `fstab`, `user-dirs.dirs` and `bookmarks` must agree. Once you get it right take copies of the config files so it is easy to repeat if you reinstall.

Tbh, I just use the linking method these days as I do not need all the home subfolders redirected. I wouldn't redirect Downloads for example as these get moved to the appropriate place afterwards anyway.

An even simpler solution is to just bookmark the folders on the Windows partition and delete the old bookmarks.  However in that case other apps might still use the original folders.

There is another solution using `bind` mounts you might investigate but I hesitate to get into that.

**Comment by jeremypw** [comment]
Closing as a number of solutions were offered.  Feel free to raise a feature request for a specific behaviour if required.

**Comment by doobes** [comment]
You can close all my support requests.

I've moved on from EOS
